### PR TITLE
DB-5953 Allow first connection on Kerberized cluster

### PR DESCRIPTION
**Pull request by dgomezferro**
Please review also -ee PR

### CODE REVIEW

**Comment by dgomezferro** [comment]
@wtang16 no, I don't think we have any kerberos ITs and it wouldn't be easy to do. I think it should be tested as part of the QA regression runs.

File: db-engine/src/main/java/com/splicemachine/db/iapi/services/property/PropertyUtil.java [hunk 1]
```diff
@@ -92,7 +92,8 @@
 		"derby.language.logQueryPlan",
 		Property.LOCKS_ESCALATION_THRESHOLD,
 		Property.DATABASE_PROPERTIES_ONLY,
-		Property.DEFAULT_CONNECTION_MODE_PROPERTY
+		Property.DEFAULT_CONNECTION_MODE_PROPERTY,
+		Property.AUTHENTICATION_BUILTIN_ALGORITHM
```
Comment (yxia92) [inline db-engine/src/main/java/com/splicemachine/db/iapi/services/property/PropertyUtil.java]: I don't quite understanding the relationship between adding this property and the main fix to allow first connection to go through without Kerberos authentication. Could you elaborate a bit?

Comment (dgomezferro) [reply db-engine/src/main/java/com/splicemachine/db/iapi/services/property/PropertyUtil.java]: The first internal connection follows a different path from normal external connections. Usually the authentication algorithm is set on that other path, so for the first internal connection it wasn't set. By setting this here I make sure there's a suitable default for that first connection.

Comment (yxia92) [reply db-engine/src/main/java/com/splicemachine/db/iapi/services/property/PropertyUtil.java]: Thanks Daniel for the explanation.

**Pull request merged**